modules/calib3d/test/test_usac.cpp

```diff
@@ -0,0 +1,105 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+
+namespace opencv_test {
+   TEST(usac_Homography, accuracy) {
+        cv::RNG rng; // random generator
+
+        // generate random rotation matrix
+        cv::Vec3d vec;
+        rng.fill(vec, cv::RNG::UNIFORM, 0, 1);
+        vec = vec / cv::norm(vec) * rng.uniform(0.0f, float(2 * CV_PI));
+        cv::Mat R;
+        cv::Rodrigues(vec, R);
+
+        // generate random translation
+        cv::Vec3d t = cv::Vec3d(rng.uniform(-0.5f, 0.5f), rng.uniform(-0.5f, 0.5f), rng.uniform(1.0f, 2.0f));
+
+        // generate random calibration
+        cv::Matx33d K = cv::Matx33d::zeros();
+        K(0, 0) = rng.uniform(100, 1000);
+        K(1, 1) = rng.uniform(100, 1000);
+        K(0, 2) = rng.uniform(-100, 100);
+        K(1, 2) = rng.uniform(-100, 100);
+        K(2, 2) = 1;
+
+        // define number of points and inlier ratio
+        const int pts_size = 5000;
+        const double inlier_ratio = 0.5;
+        // compute inlier number
+        const int inl_size = static_cast<int>(inlier_ratio * pts_size);
+        const int out_size = pts_size - inl_size;
+
+        cv::Mat inliers, outliers, points3d;
+        inliers.create(2, inl_size, CV_64F);
+        rng.fill(inliers, cv::RNG::UNIFORM, 0, 1);
```

Comment:
Are `inliers` doubles? Why?

Reply:
Because I use doubles for ground truth rotation and calibration otherwise I would have error in matrix multiplication. Since it is just test files I want to have higher precision. My RANSAC is using float points. Also in new commit (in test file) I save indices of ground truth inliers as integers.